src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -227,6 +230,53 @@ void CascadiaSettings::_CreateDefaultProfiles()
 
     _profiles.emplace_back(powershellProfile);
     _profiles.emplace_back(cmdProfile);
+

```

Comment:
I would greatly prefer if you put this entire block of stuff into its own function that is something like `std::vector<TerminalApp::Profile> _CreateWslProfiles()` or `void _CreateWslProfiles(std::vector<TerminalApp::Profile>& profileStorage)`.

Then inside here, wrap the entire thing with 
```
try
{

}
CATCH_LOG();
```
since the failure to enumerate and add the WSL profiles is something that shouldn't stop the default profile creation, but should probably be logged in case someone wants to diagnose why it failed.

Finally, rework the contents of the method to follow the `wil/result.h` based style of running down the method and doing `THROW_IF_WIN32_BOOL_FALSE` on methods like `CreatePipe`, `CreateProcessW`, `PeekNamedPipe`, `GetExitCodeProcess` and checking the result code in your own way for `WaitForSingleObject` to verify it is returning according to the conditions you expect and not a failure case, throwing if it is unexpected.

e.g.
```
THROW_IF_WIN32_BOOL_FALSE(CreatePipe(&readPipe, &writePipe, &sa, 0));

STARTUPINFO si{};

....

THROW_IF_WIN32_BOOL_FALSE(CreateProcessW(...));

...
```

Reply:
I like the pass-by-reference version better, so I went with that one.

Everything is now in a try block, with a few exceptions thrown.

Reply:
I know @miniksa suggested _CreateWslProfiles(std::vector<TerminalApp::Profile>& profileStorage) - but this doesn't jive with the CppCoreGuidelines.
[F.20: For “out” output values, prefer return values to output parameters](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f20-for-out-output-values-prefer-return-values-to-output-parameters) 

@miniksa - Is this a recommendation that you guys aren't recommending/enforcing?  Just curious for future code reviews.

Reply:
@dlong11 That is for values that are not "in-out", which is what we're treating the profiles list as. http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f17-for-in-out-parameters-pass-by-reference-to-non-const

Reply:
@JBanks Sorry about that I was looking at the modified function and the name.  I should have looked closer to the calling code.

Reply:
@dlong11 No problem at all.  I appreciate you taking the time to look things over!